app/views/DiscussionMessagesView.js

```diff
@@ -0,0 +1,216 @@
+	const isMasterDetail = useSelector(state => state.app.isMasterDetail);
+
+	const [loading, setLoading] = useState(false);
+	const [discussions, setDiscussions] = useState([]);
+	const [search, setSearch] = useState([]);
+	const [isSearching, setIsSearching] = useState(false);
+
+	const { theme } = useTheme();
+	const insets = useSafeAreaInsets();
+
+	const load = async(text = '') => {
+		if (loading) {
+			return;
+		}
+
+		setLoading(true);
+		try {
+			const result = await RocketChat.getDiscussions({
+				roomId: rid,
+				offset: isSearching ? search.length : discussions.length,
+				count: API_FETCH_COUNT,
+				text
+			});
+
+			if (result.success) {
+				if (isSearching) {
+					setSearch(result.messages);
+				} else {
+					setDiscussions(result.messages);
+				}
+			}
+			setLoading(false);
+		} catch (e) {
+			log(e);
+			setLoading(false);
+		}
+	};
+
+	const onSearchChangeText = debounce(async(text) => {
+		setIsSearching(true);
+		await load(text);
+	}, 300);
+
+	const onCancelSearchPress = () => {
+		setIsSearching(false);
+		load();
+	};
+
+	const onSearchPress = () => {
+		setIsSearching(true);
+	};
+
+	const setHeader = () => {
+		if (isSearching) {
+			const headerTitlePosition = getHeaderTitlePosition({ insets, numIconsRight: 1 });
+			return {
+				headerTitleAlign: 'left',
+				headerLeft: () => (
+					<HeaderButton.Container left>
+						<HeaderButton.Item
+							iconName='close'
+							onPress={onCancelSearchPress}
+						/>
+					</HeaderButton.Container>
+				),
+				headerTitle: () => (
+					<SearchHeader
+						placeholder='Search Messages'
+						onSearchChangeText={onSearchChangeText}
+						theme={theme}
+						testID='discussion-messages-view-search-header'
+					/>
+				),
+				headerTitleContainerStyle: {
+					left: headerTitlePosition.left,
+					right: headerTitlePosition.right
+				},
+				headerRight: () => null
+			};
+		}
+
+		const options = {
+			headerLeft: () => (
+				<HeaderBackButton
+					labelVisible={false}
+					onPress={() => navigation.pop()}
+					tintColor={themes[theme].headerTintColor}
+				/>
+			),
+			headerTitleAlign: 'center',
+			headerTitle: I18n.t('Discussions'),
+			headerTitleContainerStyle: {
+				left: null,
+				right: null
+			}
+		};
+
+		if (isMasterDetail) {
+			options.headerLeft = () => <HeaderButton.CloseModal navigation={navigation} />;
+		}
+
+		options.headerRight = () => (
+			<HeaderButton.Container>
+				<HeaderButton.Item iconName='search' onPress={onSearchPress} />
+			</HeaderButton.Container>
+		);
+		return options;
+	};
+
+	useEffect(() => {
+		load();
+	}, []);
+
+	useEffect(() => {
+		const options = setHeader();
+		navigation.setOptions(options);
+	}, [navigation, isSearching]);
+
+
+	const onDiscussionPress = debounce((item) => {
+		navigation.push('RoomView', {
+			rid: item.drid, prid: item.rid, name: item.msg, t: 'p'
+		});
+	}, 1000, true);
+
+	const renderItem = ({ item }) => (
+		<Message
```

Comment:
The user would expect to navigate to the discussion by tapping the whole item, not just the button.